Actions/VerifyPRChanges/VerifyPRChanges.ps1

```diff
@@ -0,0 +1,37 @@
+Param(
+    [Parameter(HelpMessage = "Base commit of the PR", Mandatory = $true)]
+    [string] $baseSHA,
+    [Parameter(HelpMessage = "Head commit of the PR", Mandatory = $true)]
+    [string] $headSHA,
+    [Parameter(HelpMessage = "The name of the repository the PR is going to", Mandatory = $true)]
+    [string] $prBaseRepository,
+    [Parameter(HelpMessage = "The name of the repository the PR is coming from", Mandatory = $true)]
+    [string] $prHeadRepository,
+    [Parameter(HelpMessage = "The URL of the GitHub API", Mandatory = $true)]
+    [string] $githubApiUrl
+
+)
+
+$ErrorActionPreference = "STOP"
+Set-StrictMode -version 2.0
+if ($prHeadRepository -ne $prBaseRepository) {
+    $headers = @{             
+        "Authorization" = 'token ${{ secrets.GITHUB_TOKEN }}'
+        "Accept" = "application/vnd.github.baptiste-preview+json"
+    }
+    $url = "$($githubApiUrl)/repos/$($prBaseRepository)/compare/$baseSHA...$headSHA"
+    $response = Invoke-WebRequest -UseBasicParsing -Headers $headers -Uri $url | ConvertFrom-Json
+    Write-Host "Files Changed:"
+    $response.files | ForEach-Object {
+      $filename = $_.filename
+      Write-Host "- $filename $_.status"
+      $extension = [System.IO.Path]::GetExtension($filename)
+      $name = [System.IO.Path]::GetFileName($filename)
+      if ($extension -eq '.ps1' -or $extension -eq '.yaml' -or $extension -eq '.yml' -or $name -eq "CODEOWNERS") {
```

Comment:
There aren't any yet, but what about `.psm1` files?

Reply:
We will need to allow .yml or .yaml files in other directories in the repo - and also .ps1 files.
I think we should disallow these files in .github (and subfolders) and in .AL-Go folders (where we automatically will run them)

Example: Our upcoming Test Task Recorder will deliver tests in .yaml format and also the ALDoc tool will produce documentation in .yaml format.

Reply:
Sounds like a separate task :)

Reply:
Yeah OK - we can include that in the PR for those things.